### PR TITLE
Storage: Add 'Bucket.location_type' property.

### PR DESCRIPTION
**Pull request by tseaver**
- Add named constants for allowed 'storage_class' values.
- Un-deprecate STANDARD storage class.
- Docs-deprecate storage classes implying location type.
- Add 'Bucket.location_type' property.
- Add named constants for allowed 'location_type' values.

### CODE REVIEW

**Comment by tseaver** [comment]
Storage Kokoro job fails due to flaky systest, #8552.

**Comment by JesseLovelace** [comment]
Thank you! Please hold off on merging until notified. Current prod rollout is planned for July 8

**Comment by tseaver** [comment]
Force-pushed to pick up the systest fix from #8617.

**Comment by tseaver** [comment]
@crwilcox, @busunkim96 Can you look into why the CLA bot is not reporting status here?

**Comment by tseaver** [comment]
@JesseLovelace Please ping the PR when the back-end rollout is complete.

File: storage/google/cloud/storage/bucket.py [hunk 1]
```diff
@@ -1377,6 +1435,38 @@ def location(self, value):
...
+
+        :rtype: str or ``NoneType``
+        :returns:
+            If set, one of :attr:`MULTI_REGION_LOCATION_TYPE`,
+            :attr:`REGION_LOCATION_TYPE`, or :attr:`DUAL_REGION_LOCATION_TYPE`,
+            else ``None``.
+        """
+        return self._properties.get("locationType")
+
+    @location_type.setter
```
Comment (JesseLovelace) [inline storage/google/cloud/storage/bucket.py]: Location type is actually a read-only field and updates to it are ignored by the back-end, should it have a setter here?

Comment (tseaver) [reply storage/google/cloud/storage/bucket.py]: The setter provides the only way to create a new bucket with a non-default `location_type`, e.g.:

```python
from google.cloud.storage.bucket import Bucket

bucket = client.bucket("<BUCKET_NAME_HERE>")
bucket.location_type = Bucket.DUAL_REGION_LOCATION_TYPE
bucket.create()
```

Comment (tseaver) [reply storage/google/cloud/storage/bucket.py]: Do we need to add `location_type` as a parameter for `Bucket.create`, and just remove the `location_type` setter?  For the `location` property, we left the setter but emit a deprecation warning, suggesting to use the `location` parameter to `Bucket.create`, but that was for backward-compatibility.

Comment (JesseLovelace) [reply storage/google/cloud/storage/bucket.py]: The field is actually ignored by the backend even on create

Comment (tseaver) [reply storage/google/cloud/storage/bucket.py]: `location_type` setter dropped in 8be3cb1.

**Comment by JesseLovelace** [comment]
Thanks!

Backend rollout is complete, this can be merged and released

**Pull request merged**